### PR TITLE
oneAPI interface tests for Linear and RBF kernels

### PR DESCRIPTION
**Pull request by vlad-nazarov**

### CODE REVIEW

File: cpp/oneapi/dal/algo/rbf_kernel/BUILD [hunk 1]
```diff
@@ -29,23 +29,32 @@ dal_test_suite(
...
-    host_tests = [
+    tests = [
         ":cpu_tests",
-    ],
-    dpc_tests = [
         ":gpu_tests",
+        ":interface_tests",
     ],
-)
+)
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/rbf_kernel/BUILD:60]: new line

File: cpp/oneapi/dal/algo/linear_kernel/test/batch.cpp [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        }
+    }
+
+    void check_result_values(double scale,
+                             double shift,
+                             const table& x_data,
+                             const table& y_data,
+                             const table& result_values) {
+        auto reference = compute_reference(scale, shift, x_data, y_data);
+        const double tol = te::get_tolerance<Float>(1e-2, 1e-2);
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/linear_kernel/test/batch.cpp:89]: How this tolerance is determined?

File: cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            check_result_values(sigma, x_data, y_data, result_values);
+        }
+    }
+
+    void check_result_values(double sigma,
+                             const table& x_data,
+                             const table& y_data,
+                             const table& result_values) {
+        auto reference = compute_reference(sigma, x_data, y_data);
+        const double tol = te::get_tolerance<Float>(1e-2, 1e-2);
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp:85]: How this tolerance is determined?

**Review by rlnx — COMMENTED** [review]
In general looks good to me

File: cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        const double diff = la::l_inf_norm(reference, la::matrix<double>::wrap(result_values));
+        CHECK(diff < tol);
+    }
+
+    la::matrix<double> compute_reference(double sigma, const table& x_data, const table& y_data) {
```
Comment (PetrovKP) [inline cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp:90]: I would suggest using a exactly formula without gemm: exp(-gamma * ||x - y||)

See reference in new_ts. 

Gemm appears as a result of not exactly the formula: exp(-gamma * (||x||  - **2(x,y)** + ||y||))

**Review by PetrovKP — COMMENTED** [review]
Are these tests run on DPC interfaces? gpu?

**Review by PetrovKP — APPROVED** [review]

File: cpp/oneapi/dal/algo/linear_kernel/test/batch.cpp (hunk 2 above)
Comment (vlad-nazarov) [reply cpp/oneapi/dal/algo/linear_kernel/test/batch.cpp]: Aligned tolerance with the another tests. In kernels tests is better to use the Normal generator. Perhaps we should wait for merge to the master.

File: cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp (hunk 3 above)
Comment (vlad-nazarov) [reply cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp]: Aligned tolerance with the another tests. In kernels tests is better to use the Normal generator. Perhaps we should wait for merge to the master.

**Comment by vlad-nazarov** [comment]
@rlnx, update tolerance as we discussed offline. Can you review again?

File: cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        return te::table_id::homogen<Float>();
+    }
+
+    void general_checks(const te::dataframe& x_data,
+                        const te::dataframe& y_data,
+                        double sigma,
+                        const te::table_id& x_data_table_id,
+                        const te::table_id& y_data_table_id) {
+        CAPTURE(sigma);
+        const table x = x_data.get_table(x_data_table_id);
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp:49]:
```suggestion
        const table x = x_data.get_table(this->get_policy(), x_data_table_id);
```

File: cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        const table y = y_data.get_table(y_data_table_id);
```
Comment (rlnx) [inline cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp:50]:
```suggestion
        const table y = y_data.get_table(this->get_policy(), y_data_table_id);
```

Comment (rlnx) [reply cpp/oneapi/dal/algo/rbf_kernel/test/batch.cpp:50]: The same for linear, sorry missed this when reviewed first time

**Comment by rlnx** [comment]
@vlad-nazarov, please make the same fixes for `get_table` in linear kernel. The reason is without policy `get_table` return data allocated on host, but for DPC++ allocations must be USM-based.

**Review by rlnx — APPROVED** [review]

**Pull request merged**